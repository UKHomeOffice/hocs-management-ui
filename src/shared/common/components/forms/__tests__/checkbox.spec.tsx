import React from 'react';
import { render, shallow } from 'enzyme';
import Checkbox from '../checkbox';

describe('Form checkbox component', () => {
    it('should render with value true', () => {
        expect(
            render(<Checkbox label="Label1-ticked" name="Name1" updateState={() => null} value="true" />)
        ).toMatchSnapshot();
    });

    it('should render with value false', () => {
        expect(
            render(<Checkbox label="Label2-not-ticked" name="Name2" updateState={() => null} value="false" />)
        ).toMatchSnapshot();
    });

    it('should execute callback on change, new value is true if currently true', () => {
        const mockCallback = jest.fn();
        const fieldName = 'checkbox-one';
        const event = { target: { name: fieldName, value: 'Text value' } };
        const wrapper = shallow(
            <Checkbox name={fieldName} updateState={mockCallback} value="true" />
        );
        mockCallback.mockReset();
        wrapper.find('input').simulate('change', event);
        expect(mockCallback).toHaveBeenCalledTimes(1);
        expect(mockCallback).toHaveBeenCalledWith({ name: fieldName, value: 'false' });
    });

    it('should execute callback on change, new value is true if currently false', () => {
        const mockCallback = jest.fn();
        const fieldName = 'checkbox-one';
        const event = { target: { name: fieldName, value: 'Text value' } };
        const wrapper = shallow(
            <Checkbox name={fieldName} updateState={mockCallback} value="false" />
        );
        mockCallback.mockReset();
        wrapper.find('input').simulate('change', event);
        expect(mockCallback).toHaveBeenCalledTimes(1);
        expect(mockCallback).toHaveBeenCalledWith({ name: fieldName, value: 'true' });
    });

});
