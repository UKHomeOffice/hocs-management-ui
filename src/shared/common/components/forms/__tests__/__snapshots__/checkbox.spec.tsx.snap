// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form checkbox component should render with value false 1`] = `
<div
  class="govuk-form-group"
>
  <label
    class="govuk-label govuk-label--s"
    for="Name2"
    id="Name2-label"
  >
    Label2-not-ticked
  </label>
  <input
    id="Name2"
    name="Name2"
    type="checkbox"
  />
</div>
`;

exports[`Form checkbox component should render with value true 1`] = `
<div
  class="govuk-form-group"
>
  <label
    class="govuk-label govuk-label--s"
    for="Name1"
    id="Name1-label"
  >
    Label1-ticked
  </label>
  <input
    checked=""
    id="Name1"
    name="Name1"
    type="checkbox"
  />
</div>
`;
